R/zchunk_LA126.distribution.R

```diff
@@ -36,123 +39,166 @@ module_energy_LA126.distribution_DISABLED <- function(command, ...) {
+
+    # ELECTRICITY TRANSMISSION AND DISTRIBUTION
+      # Filtering energy balance by "net_electricity distribution" sector and joining with electricity generation output (i.e., L126.out_EJ_R_elecownuse_F_Yh)
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) %>%
+      filter(sector == "net_electricity distribution") %>%
+      mutate(sector = substr(sector, 5, 30)) %>% # Renaming sector as "electricity distribution"
+      group_by(GCAM_region_ID, sector, fuel, year) %>%
+      summarise(value_a = sum(value)) %>%
+      left_join_error_no_match( # Joining electricity generation output (L126.out_EJ_R_elecownuse_F_Yh, wherein electricity ownuse was subtracted from electricity generation)
+        select(
+          ungroup(L126.out_EJ_R_elecownuse_F_Yh), # Need to ungroup to deselect sector and fuel
+          -sector, -fuel),
+        by = c("GCAM_region_ID", "year")) %>%
+      mutate(value_b = value - value_a) %>% # Creating values for table, L126.out_EJ_R_electd_F_Yh (i.e. ouput), by subtracting electricity generation (without ownuse) by transmission and distribution consumption
+      mutate(value_c = value / value_b) -> # Creating values for table, L126.IO_R_electd_F_Yh, by dividing input by output
+      c
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      # Table c is separated to create the final tables
+    c %>% select(GCAM_region_ID, sector, fuel, year, value) -> L126.in_EJ_R_electd_F_Yh
+    c %>% select(GCAM_region_ID, sector, fuel, year, value = value_b) -> L126.out_EJ_R_electd_F_Yh
+    c %>% select(GCAM_region_ID, sector, fuel, year, value = value_c) -> L126.IO_R_electd_F_Yh
+
+    # GAS PIPELINE
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) %>%
+      filter(sector == "net_gas pipeline") %>%
+      mutate(sector = substr(sector, 5, 30)) %>% # Renaming sector as "gas pipeline"
+      group_by(GCAM_region_ID, sector, fuel, year) %>%
+      summarise(value_a = sum(value)) %>%
+      left_join( # Joining L122.out_EJ_R_gasproc_F_Yh, but first converting to long form, stripping years of X's, and summing by GCAM region ID and year
+        summarise(
+          group_by(
+            mutate(
+              gather(L122.out_EJ_R_gasproc_F_Yh, year, value, -GCAM_region_ID, -sector, -fuel),
+              year = as.integer(substr(year, 2, 5))),
```

Comment:
You do not want to see my latest code then! My goal was to write one long stream without a single break. I thought that was the most efficient because it has the fewest lines.